include/leo_s3_auth.hrl

```diff
@@ -51,6 +51,8 @@
           requested_uri = <<>> :: binary(),
           query_str     = <<>> :: binary(),
           sub_resource  = <<>> :: binary(), %% [?acl" | "?location" | "?logging" | "?torrent"]
+          sign_ver      = v2   :: atom(),   %% [v2 | v4]
+          req                  :: cowboy_req:req(),
           amz_headers   = []   :: list()
          }).
 
```

Comment:
Erlang allow users to create user defined type to make static analysis more robust like this.
```erlang
-type aws_sign_ver() ::  v2 | v4.
```

Please apply this method as possible.